### ISSUE TITLE
Pan view image class filtering

### ISSUE THREAD

**Issue by tomaszg7** [BestImageViewer/geeqie#612]
Continuing problems related to video support in Geeqie (started in https://github.com/BestImageViewer/geeqie/issues/510) it might be good to have an option in Pan view to filter by the class of image (RAW, image, video).

**Comment by tomaszg7** [comment]
Also a way to start playback of a video file from Pan view would be handy (e.g. by putting "play" command into context menu).

**Comment by tomaszg7** [comment]
Tried to solve the first issue, attaching patch. I didn't want to change pan view behaviour, so it displays only one instance from each group. Thus it can miss some class, e.g. if IMAGE and RAWIMAGE are grouped together, only one will be displayed. Maybe some fallback might be reasonable (i.e. if image class was not matched, look for other representative of the group matching the filter setting). In the current state, maybe UNKNOWN and METADATA checkbox should be disabled.

I had to un-`static` `format_class_list` in `preferences.c` to reuse it, adding corresponding `extern` in `typedef.h`. I don't know if it is a correct approach.

Also, changed properties of `filter_mode_combo` not to expand. I think it looks better this way.

[geeqie-panview-filter-by-class.patch.gz](https://github.com/BestImageViewer/geeqie/files/2066433/geeqie-panview-filter-by-class.patch.gz)
